### PR TITLE
added cloudfront service (get_distribution, list_distributions, list_distributions_all)

### PR DESCRIPTION
**Pull request by andrey-smirnov**
CloudFront GetDistribution and ListDistributions API calls implemented
@aporubov @motobob

### CODE REVIEW

File: src/erlcloud_cloudfront.erl [hunk 1]
```diff
@@ -0,0 +1,243 @@
+-module(erlcloud_cloudfront).
+
+-include_lib("xmerl/include/xmerl.hrl").
+
+%% Library initialization.
+-export([configure/2, configure/3, new/2, new/3]).
+
+-export([list_distributions_all/0, list_distributions_all/1,
```
Comment (andrey-smirnov) [inline src/erlcloud_cloudfront.erl]: well, all EC2 module filled with _all functions. Do you propose just to remove this "get all results" stuff and pagination for user?

File: src/erlcloud_cloudfront.erl [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+
+
+-spec list_distributions(integer(), string()) -> ok_error(proplist(), string()).
+list_distributions(MaxResults, Marker) ->
+    list_distributions(MaxResults, Marker, erlcloud_aws:default_config()).
+
+-spec list_distributions(integer(), string(), aws_config()) -> ok_error(proplist(), string()).
+list_distributions(MaxResults, Marker, Config) when
+    is_list(Marker) orelse Marker =:= undefined ->
+    Params = [{"MaxItems", MaxResults}] ++ case Marker of
```
Comment (andrey-smirnov) [inline src/erlcloud_cloudfront.erl]: where exactly this tons of wrappers, could you point it out?

File: src/erlcloud_cloudfront.erl [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+            E
+    end.
+
+
+-spec get_distribution(string()) -> ok_error(proplist()).
+get_distribution(Id) ->
+    get_distribution(Id, erlcloud_aws:default_config()).
+
+-spec get_distribution(string(), aws_config()) -> ok_error(proplist()).
+get_distribution(Id, Config) ->
```
Comment (motobob) [inline src/erlcloud_cloudfront.erl]: rest have the guards for `is_record()` so please add it here too.

File: src/erlcloud_cloudfront.erl [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+       {certificate, "Certificate", optional_text},
+       {certificate_source, "CertificateSource", optional_text},
+       {cloudfront_default_certificate, "CloudFrontDefaultCertificate", optional_text},
+       {iam_certificate_id, "IAMCertificateId", optional_text},
+       {minimum_protocol_version, "MinimumProtocolVersion", optional_text},
+       {ssl_support_method, "SSLSupportMethod", optional_text}
+      ], Node).
+
+
+flush_undefined(L) ->
```
Comment (motobob) [inline src/erlcloud_cloudfront.erl]: i do not see a need for a fun named like this. use filter direct in below please.

**Pull request merged**